server/store.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-zoom/server/zoom"
+)
+
+const (
+	zoomStateKeyPrefix = "zoomuserstate"
+	postMeetingKey     = "post_meeting_"
+)
+
+func (p *Plugin) storeOAuthUserInfo(info *zoom.OAuthUserInfo) error {
+	config := p.getConfiguration()
+
+	encryptedToken, err := encrypt([]byte(config.EncryptionKey), info.OAuthToken.AccessToken)
+	if err != nil {
+		return errors.Wrap(err, "could not encrypt OAuth token")
+	}
+	info.OAuthToken.AccessToken = encryptedToken
+
+	encoded, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+
+	if err := p.API.KVSet(zoomTokenKey+info.UserID, encoded); err != nil {
+		return err
+	}
+
+	if err := p.API.KVSet(zoomTokenKeyByZoomID+info.ZoomID, encoded); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) fetchOAuthUserInfo(tokenKey, userID string) (*zoom.OAuthUserInfo, error) {
+	encoded, appErr := p.API.KVGet(tokenKey + userID)
+	if appErr != nil || encoded == nil {
+		return nil, errors.New("must connect user account to Zoom first")
+	}
+
+	var info zoom.OAuthUserInfo
+	if err := json.Unmarshal(encoded, &info); err != nil {
+		return nil, errors.New("could not to parse OAauth access token")
+	}
+
+	return &info, nil
+}
+
+func (p *Plugin) disconnectOAuthUser(userID string) error {
+	encoded, err := p.API.KVGet(zoomTokenKey + userID)
+	if err != nil {
+		return errors.Wrap(err, "could not find OAuth user info")
+	}
+
+	var info zoom.OAuthUserInfo
+	if err := json.Unmarshal(encoded, &info); err != nil {
+		return errors.Wrap(err, "could not decode OAuth user info")
+	}
+
+	errByMattermostID := p.API.KVDelete(zoomTokenKey + userID)
+	errByZoomID := p.API.KVDelete(zoomTokenKeyByZoomID + info.ZoomID)
+	if errByMattermostID != nil {
+		return errByMattermostID
+	}
+	if errByZoomID != nil {
+		return errByZoomID
+	}
+	return nil
+}
+
+// storeUserState stores the user state with the corresponding channelID
+func (p *Plugin) storeUserState(userID string, channelID string) error {
+	state := fmt.Sprintf("%s_%s", zoomStateKeyPrefix, userID)
```

Comment:
By using a fixed prefix here instead, I think we can avoid having a bunch of excess state entries in the KV store due to incomplete connection attempts from users.

Reply:
To avoid this, the best way to do it is with `KVSetWithExpiry` (a few functions below). I would put a TTL of 5 minutes here.